docs/IfCanAccess.md

```diff
@@ -5,29 +5,157 @@ title: "IfCanAccess"
+
+For instance, if the user has the following permissions:
+
+```jsx
+console.log(await authProvider.getPermissions())
+// [
+//     { action: ["read", "create", "edit", "export"], resource: "companies" },
+//     ...
+// ];
+```
+
+To display the `ExportButton` in a `CompanyList` component, you would use:
+
+```jsx
+<IfCanAccess action="export">
+    <ExportButton />
+</IfCanAccess>
+```
+
+## `resource`
+
+By default, `<UseCanAccess>` uses the current resource (from the `ResourceContext`) to check permissions. You can override this behavior by passing the `resource` prop:
+
+```jsx
+<IfCanAccess action="export" resource="companies">
+    <ExportButton />
+</IfCanAccess>
+```
+
+## `record`
+
+RBAC allows to specify [record-level permissions](./AuthRBAC.md#record-level-permissions). These permissions are triggered when you specify the `record` prop.
+
+For example, let's say a user has the permission to edit a company only if the company is in the same group as the user:
+
+```jsx
+console.log(await authProvider.getPermissions())
+// [
+    // { action: 'edit', resource: "companies', record: { group: 'middle_east' } },
+// ];
+```
+
+To display the `EditButton` in a `CompanyShow` component, you would use:
+
+```jsx
+const EditCompanyButton = () => {
+    const record = useRecordContext();
+    return (
+        <IfCanAccess action="edit" record={record}>
+            <EditButton />
+        </IfCanAccess>
+    );
+};
+```
+
+## `fallback`
+
+`ra-rbac` shows a Not Found page when users try to access a page they don't have the permissions for. It is considered good security practice not to disclose to a potentially malicious user that a page exists if they are not allowed to see it.
+
+However, should you prefer to show an Access Denied screen in those cases, you can do so by specifying a `fallback` component in `<IfCanAccess>`:
+
+```tsx
+// in src/posts/PostCreate.tsx
+import { Create, SimpleForm, TextInput } from 'react-admin';
+import { IfCanAccess } from '@react-admin/ra-rbac';
+import { Navigate } from 'react-router-dom';
+
+export const PostCreate = () => (
+    <IfCanAccess action="create" fallback={<Navigate to="/access-denied" />}>
+        <Create>
+            <SimpleForm>
+                <TextInput source="title" />
+            </SimpleForm>
+        </Create>
+    </IfCanAccess>
+);
+```
+
+**Tip**: This example uses a `Navigate` component to redirect to a custom page because you cannot use a `Redirect` component in this context. The `IfCanAccess` component uses a render prop, and `Redirect` only works in the render method of a component.
+
+Note that you if you use the `fallback` prop for a CRUD page (Create, Edit, List, Show) as above, you must use the `<Resource>` component from `react-admin` rather than the one from `ra-rbac`. This is because `ra-rbac` already does the access control check, and would redirect to the Not Found page before the fallback component is rendered.
```

Comment:
```suggestion
Note that if you use the `fallback` prop for a CRUD page (Create, Edit, List, Show) as above, you must use the `<Resource>` component from `react-admin` rather than the one from `ra-rbac`. This is because `ra-rbac` already does the access control check, and would redirect to the Not Found page before the fallback component is rendered.
```